pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -556,6 +556,11 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 		}
 		pod, containerSpec = restoredPod, restoredContainer
 	}
+
+	if containerSpec == nil {
+		return fmt.Errorf("failed to get container %q in pod %q", containerName, format.Pod(pod))
```

Comment:
How about including `containerID` in the error message?  In #48173 we see that this value is non-empty, so it would provide an additional clue.